arch/x86/kvm/x86.c

```diff
@@ -7077,6 +7086,13 @@ static int kvm_vm_ioctl_set_msr_filter(struct kvm *kvm,
 	int r;
 	u32 i;
 
+	/*
+	 * The pKVM hypervisor will ignore the MSR filter configured by the host
+	 * user space VMM.
+	 */
+	if (enable_pkvm)
+		return -EOPNOTSUPP;
+
```

Comment:
If we prefer to disable this ioctl, we may also need to disable some extension reported by the KVM, e.g., KVM_CAP_X86_USER_SPACE_MSR as some user space VMM (e.g., QEMU) may expect to get a successful return from ioctl if the KVM reports this extension.

But seems the user space VMM can still use this ioctl to filter out the host emulated MSR, although it cannot impact the MSR emulated by the pKVM. That is similar to the TDX which also doesn't disable the MSR filter. So it seems we also don't need to disable this ioctl?

Reply:
> So, perhaps ideally we could return -EOPNOTSUPP for pKVM-emulated MSRs, while let userspace use MSR filters for host-emulated ones. But yeah, we don't need to implement that now (especially if it would complicate the code).

Yeah, agree. We can just simply remove the check here for now until we need to implement for that.